deployment/vm/manager.go

```diff
@@ -94,9 +94,32 @@ func (m *manager) Create(stemcell bistemcell.CloudStemcell, deploymentManifest b
 		return nil, bosherr.WrapErrorf(err, "Getting resource pool for job '%s'", jobName)
 	}
 
-	agentID, err := m.uuidGenerator.Generate()
+	agentID, found, err := m.vmRepo.FindCurrentAgentId()
 	if err != nil {
-		return nil, bosherr.WrapError(err, "Generating agent ID")
+		return nil, bosherr.WrapError(err, "Finding currently agent id of deployed vm")
+	}
+
+	if !found {
```

Comment:
I don't understand this? If no `agentID` is found then we generate a new ID and move on??